src/renderer/caching_shaper.rs

```diff
@@ -48,101 +40,159 @@ struct Asset;
 
 const DEFAULT_FONT_SIZE: f32 = 14.0;
 
+#[derive(Clone)]
+pub struct ExtendedFontFamily {
+    pub fonts: Vec<SkriboFont>,
+}
+
+impl ExtendedFontFamily {
+    pub fn new() -> ExtendedFontFamily {
+        ExtendedFontFamily { fonts: Vec::new() }
+    }
+
+    pub fn add_font(&mut self, font: SkriboFont) {
+        self.fonts.push(font);
+    }
+
+    pub fn get(&self, props: Properties) -> Option<&Font> {
+        for handle in &self.fonts {
+            let font = &handle.font;
+            let properties = font.properties();
+
+            if properties.weight == props.weight && properties.style == props.style {
+                return Some(&font);
+            }
+        }
+
+        if let Some(handle) = &self.fonts.first() {
+            return Some(&handle.font);
+        }
+
+        None
+    }
+
+    pub fn to_family(self) -> FontFamily {
+        let mut new_family = FontFamily::new();
```

Comment:
I think this could be replaced by `FontFamily { fonts: self.fonts }` which would be faster than creating a new font family and adding each one.

Reply:
yeah the reason i did it like that is that skribo for some reason has fonts set to private... it sucks

Reply:
Ah I see. So this line means public only for the crate? I'm not familiar with Rust's access rules https://github.com/linebender/skribo/blob/master/src/collection.rs#L16